patches/server/0731-Add-System.out.println-catcher.patch

```diff
@@ -0,0 +1,105 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: underscore11code <minecrafter11mrt@gmail.com>
+Date: Fri, 23 Jul 2021 23:01:42 -0700
+Subject: [PATCH] Add System.out.println catcher
+
+
+diff --git a/src/main/java/io/papermc/paper/logging/SysoutCatcher.java b/src/main/java/io/papermc/paper/logging/SysoutCatcher.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..c3d37945c7d0e23b36000edd6fde4525adf87159
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/logging/SysoutCatcher.java
+@@ -0,0 +1,73 @@
++package io.papermc.paper.logging;
++
++import org.bukkit.Bukkit;
++import org.bukkit.plugin.Plugin;
++import org.bukkit.plugin.java.JavaPlugin;
++
++import java.io.PrintStream;
++
++public final class SysoutCatcher {
++    private static final String PREFIX = "[SOUT] ";
++    private final StackWalker stackWalker = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
++
++    public SysoutCatcher() {
++        // Listen to System.out.println
++        PrintStream wrappedOut = new PrintStream(System.out) {
++            @Override
++            public void println(String line) {
++                Class<?> clazz = stackWalker.getCallerClass();
++                try {
++                    JavaPlugin plugin = JavaPlugin.getProvidingPlugin(clazz);
++
++                    // Instead of just printing the message, send it to the plugin's logger
++                    plugin.getLogger().info(PREFIX + line);
++                    nag(plugin);
++                } catch (IllegalArgumentException e) {
++                    // If anything happens, the calling class doesn't exist, there is no JavaPlugin that "owns" the calling class, etc
++                    // Just print out normally, with some added information
++                    Bukkit.getLogger().info(String.format("[%s]" + PREFIX + " %s", clazz.getSimpleName(), line));
```

Comment:
```suggestion
+                    Bukkit.getLogger().info(String.format("[%s]" + PREFIX + " %s", clazz.getName(), line));
```
Would it be possible to use the full class name instead of the simple name? It would help with debugging considering it is much easier to locate a fully-qualified class name.

Reply:
Yeah I suppose that would probably be better.